source/features/review-comments-hidden-indicator.tsx

```diff
@@ -0,0 +1,131 @@
+import './review-comments-hidden-indicator.css';
+import React from 'dom-chef';
+import select from 'select-dom';
+import features from '../libs/features';
+import observeEl from '../libs/simplified-element-observer';
+import {comment as commentIcon} from '../libs/icons';
+
+const SELECTOR_COMMENT_CONTAINER = 'tr.inline-comments';
+const SELECTOR_CUSTOM_TOGGLE = 'tr.refined-toggle-comments';
+const SELECTOR_COMMENT = '.review-comment .js-comment';
+
+let isVisible = true; // Are comments visible?
+
+const getOffsetY = (el: Element): number =>
+	el.getBoundingClientRect().top + window.pageYOffset;
+
+// Toggle comments while maintaining scroll position
+const toggleComments = (target: HTMLElement): void => {
+	const container = target.closest(SELECTOR_CUSTOM_TOGGLE)!;
+	const scrollOld = window.scrollY;
+	const offsetOld = getOffsetY(container);
+
+	select.all<HTMLInputElement>('input.js-toggle-file-notes').forEach(input => {
+		if (isVisible === input.checked) {
+			input.click();
+		}
+	});
+	isVisible = !isVisible;
+
+	const offsetNew = getOffsetY(container);
+	const scrollY = offsetNew - offsetOld + scrollOld;
+	window.scrollTo(0, scrollY);
+};
+
+const removeToggles = (): void => {
+	const toRemove = select.all(SELECTOR_CUSTOM_TOGGLE).filter(el => {
+		const next = el.nextElementSibling;
+		return !next || !next.matches(SELECTOR_COMMENT_CONTAINER);
+	});
+	toRemove.forEach(el => el.remove());
+};
+
+// Watch for comment addition/deletions and update comment count
+const countListener = (mutations: MutationRecord[]): void => {
+	const selectors = 'td.line-comments, .review-comment, .js-comments-holder';
+	for (const mutation of mutations) {
+		if (mutation.target.nodeType !== 1) {
+			continue;
+		}
+
+		const target = mutation.target as HTMLElement;
+		if (!target.matches(selectors)) {
+			continue;
+		}
+
+		const container = (mutation.target as HTMLElement).closest('tr')!;
+		const toggler = container.previousElementSibling;
+		if (toggler && toggler.matches(SELECTOR_CUSTOM_TOGGLE)) {
+			const comments = select.all(SELECTOR_COMMENT, container);
+			const counter = select('span', toggler)!;
+			counter.textContent = comments.length.toString();
+		} else {
+			removeToggles();
+		}
+
+		break;
+	}
+};
+
+const commentToggle = (count: number): JSX.Element => (
+	<tr className="refined-toggle-comments">
+		<td className="blob-num" colSpan={2}>
+			<button onClick={ev => toggleComments(ev.target as HTMLElement)}>
```

Comment:
Create a generic handler function instead of creating one inline here.

```suggestion
			<button type="button" onClick={showCommentsHandler}>
```

Basically just renamed `toggleComments` above to `showCommentsHandler` and just get the `target` in there

Reply:
TypeScript was shouting over MouseEvent, will try again :)

Reply:
Search in other features, there’s some weird type called React.MouseEvent

Reply:
Thanks. It works. Just fails in my VSCode 🎉